### PR TITLE
[INTERP] Fixed some undefined behaviour

### PR DESCRIPTION
**Pull request by ksco**
It's probably just fine, but just in case...

### CODE REVIEW

File: src/emu/x64primop.h [hunk 1]
```diff
@@ -311,7 +311,7 @@ static inline uint8_t shl8(x64emu_t *emu, uint8_t d, uint8_t s)
 		emu->df = d_shl8;
 		emu->op1.u8 = d;
 		emu->op2.u8 = s;
-		emu->res.u8 = d << s;
+		emu->res.u8 = s > 7 ? 0 : d << s;
```
Comment (ptitSeb) [inline src/emu/x64primop.h:314]: that's fine, but I think you need paranthesis, I don't think `:` as more priority than `<<` (at least it's more clear that way).

Comment (ptitSeb) [reply src/emu/x64primop.h:314]: so `(s > 7) ? 0 : (d <<s)` (I also find the space to be useless with the paranthesis, but it's just personal taste)

**Review by ptitSeb — CHANGES_REQUESTED** [review]
please add some `(` `)`

**Comment by ksco** [comment]
Will do, I'm out for dinner.

**Comment by ksco** [comment]
done.

**Review by ptitSeb — APPROVED** [review]

**Pull request merged**